File: JavaConsoleProgram/Node.java
import java.util.*;

public class Node {
    private int x;
    private int y;
    private int degree;
    private int orderVisited;
    private boolean isVisited;
    private Node nextNode = null;
    public Node(){

    }
    public Node(int x,int y){
        this.x = x;
        this.y = y;
    }
    public void setX(int x){
        this.x = x;
    }
    public void setY(int y){
        this.y = y;
    }
    public void setDegree(int degree){
        this.degree = degree;
    }
    public void setOrderVisited(int order){
        this.orderVisited = order;
    }
    public void setIsVisited(Boolean visited){
        this.isVisited = visited;
    }
    public void setNextNode(Node next){
        this.nextNode = next;

        
    }    
    public int getX(){
        return this.x;
    }
    public int getY(){
        return this.y;
    }
    public int getDegree(){
        return this.degree;
    }
    public int getOrderVisited(){
        return this.orderVisited;
    }
    public boolean getIsVisited(){
        return this.isVisited;
    }
    public Node getNextNode(){
        return this.nextNode;
    }
}

File: cppConsoleProgram/node.cpp
#include "node.h"
#include <iostream>

Node::Node() {
	this->x = 0;
	this->y = 0;
	this->degree = 0;
	this->orderVisited = 0;
	this->isVisited = false;
	this->nextNode = NULL;
}

Node::Node(int x,int y) {
	this->x = x;
	this->y = y;
	this->degree = 0;
	this->orderVisited = 0;
	this->isVisited = false;
	this->nextNode = NULL;
}

void Node::setDegree(int degree) {
	this->degree = degree;
}
void Node::setX(int x) {
	this->x = x;
}
void Node::setY(int y) {
	this->y = y;
}
void Node::setOrderVisited(int order) {
	this->orderVisited = order;
}
void Node::setIsVisited(bool isVisited) {
	this->isVisited = isVisited;
}
void Node::setNextNode(Node* next) {
	this->nextNode = next;
}

int Node::getX() {
	return this->x;
}
int Node::getY() {
	return this->y;
}
int Node::getDegree() {
	return this->degree;
}
int Node::getOrderVisited() {
	return this->orderVisited;
}
bool Node::getIsVisited() {
	return this->isVisited;
}
Node* Node::getNextNode() {
	return this->nextNode;
}